File: LeetCode/1-100/11. 盛最多水的容器.java
class Solution {
    public int maxArea(int[] height) {
 int left = 0, right = height.length - 1;
        int area = 0;
        while (left < right) {
            int len = right - left;
            int tail = height[left] > height[right] ? height[right--] : height[left++];
            if (len * tail > area) area = len * tail;
        }
        return  area;
    }
}

File: LeetCode/1-100/11. 盛最多水的容器.go
func maxArea(height []int) int {
    n := len(height)
    left,right := 0,n - 1
    res := 0
    for left < right {
        curLen := right -left
        tail := 0
        if height[left] > height[right] {
            tail = height[right]
            right--
        } else {
            tail = height[left]
            left++
        }
        res = max(curLen*tail,res)
    }
    return res
}


func max(a,b int) int {
    if a > b {
        return a
    } else {
        return b
    }
}